### PR TITLE
W2 - Anhquach - Create calculator view

### PR DESCRIPTION
**Pull request by at-anhquach**
#### Related issues (Optional)
Task 2

#### Description (Required)
- [x] Create Calculator View

### CODE REVIEW

File: app/src/at-anhquach/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 <resources>
     <string name="app_name">anhquach-17k3</string>
+    <string name="Add">+</string>
+    <string name="Divide">/</string>
+    <string name="Multi">*</string>
+    <string name="Sub">-</string>
+    <string name="NoteDivideZero">Cant divide for 0</string>
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/values/strings.xml]: WARNING: [Typos] "Cant" is a common misspelling; did you mean "Cannot" or "Can not" or "Can't" ?
 This check looks through the string definitions, and if it finds any words that look like likely misspellings, they are flagged.

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    </LinearLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="10dp"
+        android:gravity="center_horizontal"
+        android:orientation="horizontal">
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            android:id="@+id/btnAdd"
+            android:layout_width="80dp"
+            android:layout_height="wrap_content"
+            android:text="@string/Add" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            android:id="@+id/btnSub"
+            android:layout_width="80dp"
+            android:layout_height="wrap_content"
+            android:text="@string/Sub" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            android:id="@+id/btnMulti"
+            android:layout_width="80dp"
+            android:layout_height="wrap_content"
+            android:text="@string/Multi" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+public class CalculatorViewActivity extends AppCompatActivity implements View.OnClickListener {
+    private EditText mEdtNumberA;
+    private EditText mEdtInputB;
+    private TextView mTvOperator;
+    private TextView mTvResutl;
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java]: mTvResult

File: app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_calculator_view);
+        init();
+        onClickListener();
+    }
+
+    @Override
+    public void onClick(View view) {
+        float numA;
+        float numB;
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java]: Move `numA` and `numB` variable to line 43, 44

File: app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+                    mTvResutl.setText("");
+                }
+                break;
+            default:
+                break;
+        }
+
+    }
+
+    private void init() {
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java]: Should rename to `initViews()` method

File: app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        mEdtInputB = findViewById(R.id.edtInputB);
+        mTvOperator = findViewById(R.id.tvOperator);
+        mTvResutl = findViewById(R.id.tvResutl);
+        mBtnAdd = findViewById(R.id.btnAdd);
+        mBtnSub = findViewById(R.id.btnSub);
+        mBtnMulti = findViewById(R.id.btnMulti);
+        mBtnDiv = findViewById(R.id.btnDiv);
+    }
+
+    private void onClickListener() {
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java]: Should rename to `initListener`

File: app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        mBtnAdd.setOnClickListener(this);
+        mBtnSub.setOnClickListener(this);
+        mBtnMulti.setOnClickListener(this);
+        mBtnDiv.setOnClickListener(this);
+    }
+
+    private void convertFloattoInt() {
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java]: Should add 2 params: `convertFloatToInt(float result, String oper)`

File: app/src/at-anhquach/java/vn/asiantech/internship/IndexActivity.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+        switch (view.getId()) {
+            case R.id.btnIssue1:
+
+            case R.id.btnIssue2:
+                Intent intentMain = new Intent(IndexActivity.this,
+                        CalculatorViewActivity.class);
+                startActivity(intentMain);
+        }
+    }
+}
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/IndexActivity.java]: Add line break

File: app/src/at-anhquach/java/vn/asiantech/internship/IndexActivity.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+/**
+ * Created by anh.quach on 11/28/17.
+ */
+
+public class IndexActivity extends AppCompatActivity implements View.OnClickListener{
+    private Button mBtnIssue2;
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/IndexActivity.java]: Move to local variable

File: app/src/at-anhquach/java/vn/asiantech/internship/MainActivity.java [hunk 13]
```diff
@@ -9,6 +9,7 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/MainActivity.java]: Remove change

File: app/src/at-anhquach/res/drawable/bg_edt_input_number.xml [hunk 14]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <stroke
+        android:width="1dp"
+        android:color="@color/colorEdtBlack">
+    </stroke>
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/res/drawable/bg_edt_input_number.xml]: Should update to:
` <stroke
        android:width="1dp"
        android:color="@color/colorEdtBlack" />`

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+            android:id="@+id/btnDiv"
+            android:layout_width="80dp"
+            android:layout_height="wrap_content"
+            android:text="@string/Divide" />
+    </LinearLayout>
+
+    <TextView
+        android:id="@+id/tvResutl"
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: Rename to `tvResult`
And make sure correct UI for multi screen

File: app/src/at-anhquach/res/layout/activity_index.xml [hunk 16]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/res/layout/activity_index.xml]: Format code

File: app/src/at-anhquach/res/values/strings.xml [hunk 17]
```diff
@@ -1,3 +1,10 @@
 <resources>
     <string name="app_name">anhquach-17k3</string>
+    <string name="Add">+</string>
+    <string name="Divide">/</string>
+    <string name="Multi">*</string>
+    <string name="Sub">-</string>
+    <string name="NoteDivideZero">Cannot divide for 0</string>
+    <string name="issueName1">Issue 1</string>
+    <string name="issuename2">Issue 2</string>
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/res/values/strings.xml]: Please check convention file and correct variable in strings file

File: app/src/at-anhquach/java/vn/asiantech/internship/IndexActivity.java [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+        setContentView(R.layout.activity_index);
+        mBtnIssue2 = findViewById(R.id.btnIssue2);
+        mBtnIssue2.setOnClickListener(this);
+        }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.btnIssue1:
+
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/IndexActivity.java]: Add break

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="10dp"
+        android:gravity="center_horizontal"
+        android:orientation="horizontal">
+
+        <Button
+            android:id="@+id/btnAdd"
+            android:layout_width="80dp"
```
Comment (at-huongnguyen) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: Use width = "0dp" and layout_weight = "1"

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+        android:orientation="horizontal">
+
+        <Button
+            android:id="@+id/btnAdd"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/operator_add" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+            android:id="@+id/btnSub"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/operator_sub" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+            android:id="@+id/btnMulti"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/operator_multi" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+    </LinearLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="20dp"
+        android:gravity="center_horizontal"
+        android:orientation="horizontal">
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+            android:id="@+id/btnAdd"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:enabled="false"
+            android:text="@string/operator_add" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+            android:id="@+id/btnSub"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:enabled="false"
+            android:text="@string/operator_sub" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+            android:id="@+id/btnMulti"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:enabled="false"
+            android:text="@string/operator_multi" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java [hunk 27]
```diff
@@ -0,0 +1,163 @@
...
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.text.DecimalFormat;
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/java/vn/asiantech/internship/CalculatorViewActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.text.DecimalFormat'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml (hunk 23 above)
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml (hunk 24 above)
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml (hunk 25 above)
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/res/layout/activity_calculator_view.xml (hunk 26 above)
Comment (at-dev-support-01) [inline app/src/at-anhquach/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-anhquach/java/vn/asiantech/internship/ui/caculatorview/CalculatorViewActivity.java [hunk 28]
```diff
@@ -0,0 +1,165 @@
...
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.text.DecimalFormat;
```
Comment (at-dev-support-01) [inline app/src/at-anhquach/java/vn/asiantech/internship/ui/caculatorview/CalculatorViewActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.text.DecimalFormat'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Pull request merged**